# -*- coding: utf-8 -*-
############################################################
#
# Autogenerated by the KBase type compiler -
# any changes made here will be overwritten
#
############################################################

from __future__ import print_function
# the following is a hack to get the baseclient to import whether we're in a
# package or not. This makes pep8 unhappy hence the annotations.
try:
    # baseclient and this client are in a package
    from .baseclient import BaseClient as _BaseClient  # @UnusedImport
except:
    # no they aren't
    from baseclient import BaseClient as _BaseClient  # @Reimport


class kb_gffupload(object):

    def __init__(
            self, url=None, timeout=30 * 60, user_id=None,
            password=None, token=None, ignore_authrc=False,
            trust_all_ssl_certificates=False,
            auth_svc='https://kbase.us/services/authorization/Sessions/Login'):
        if url is None:
            raise ValueError('A url is required')
        self._service_ver = None
        self._client = _BaseClient(
            url, timeout=timeout, user_id=user_id, password=password,
            token=token, ignore_authrc=ignore_authrc,
            trust_all_ssl_certificates=trust_all_ssl_certificates,
            auth_svc=auth_svc)

    def fasta_gff_to_genome(self, params, context=None):
        """
        :param params: instance of type "FastaGFFToGenomeParams" (fasta_file
           - file containing assembled contigs/chromosomes gff_file - file
           containing gene models (_must_ contain 'gene', 'mRNA', and 'CDS')
           genome_name - becomes the name of the object workspace_name - the
           name of the workspace it gets saved to. source - Source of the
           file typically something like RefSeq or Ensembl taxon_ws_name -
           where the reference taxons are : ReferenceTaxons taxon_reference -
           if defined, will try to link the Genome to the specified taxonomy
           object release - Release or version number of the data
           (i.e.Ensembl Release 31 or Phytozome Release V11) type -
           Reference, Representative or User) -> structure: parameter
           "fasta_file" of type "File_Path", parameter "gff_file" of type
           "File_Path", parameter "genome_name" of String, parameter
           "workspace_name" of String, parameter "source" of String,
           parameter "taxon_wsname" of String, parameter "taxon_reference" of
           String, parameter "release" of String, parameter "type" of String
        :returns: instance of type "GenomeSaveResult" -> structure: parameter
           "genome_ref" of String, parameter "report_name" of String,
           parameter "report_ref" of String
        """
        return self._client.call_method(
            'kb_gffupload.fasta_gff_to_genome',
            [params], self._service_ver, context)

    def status(self, context=None):
        return self._client.call_method('kb_gffupload.status',
                                        [], self._service_ver, context)
